app/js/components/CreateEscrow.jsx

```diff
@@ -0,0 +1,82 @@
+import React, {Component} from 'react';
+import {Alert, Button, Card, CardBody, CardHeader, CardTitle, Form, FormGroup, Input, Label} from 'reactstrap';
+import PropTypes from 'prop-types';
+
+function pad(number) {
+  if (number < 10) {
+    return '0' + number;
+  }
+  return number;
+}
+
+function toInputDate(date) {
+  return date.getUTCFullYear() +
+    '-' + pad(date.getUTCMonth() + 1) +
+    '-' + pad(date.getUTCDate()) +
+    'T' + pad(date.getUTCHours()) +
+    ':' + pad(date.getUTCMinutes());
+}
+
+class CreateEscrowForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: '',
+      expiration: new Date(),
+      buyer: ''
+    };
+  }
+
+  onChange(e, field) {
+    let data = e.target.value;
+    if (field === 'expiration') {
+      data = new Date(e.target.value);
+    }
+    this.setState({[field]: data});
+  }
+
+  submit = () => {
+    // TODO add validation before submitting
+    this.props.create(this.state.buyer, this.state.value, this.state.expiration.getTime());
+  };
+
+  render() {
+    return <Card className="mt-2">
+      <CardHeader>
+        <CardTitle>Create an Escrow</CardTitle>
+      </CardHeader>
+      <CardBody>
+        <Form>
+          {this.props.error &&
+          <Alert color="danger">Error while creating the escrow: {this.props.error}</Alert>}
+          {this.props.result &&
+          <Alert color="success">Escrow receipt: <pre>{JSON.stringify(this.props.result, null, 2)}</pre></Alert>}
```

Comment:
This is already really nice, and more than we ever do so maybe we can keep this one for later but we should probably add a loading state, a tx can take a lot of time!

Reply:
Yup for sure, it was in my mental backlog. I wasn't sure how "good" we wanted the features since it's an MVP, that's why I didn't add validation yet either.